### PR TITLE
#666 - Removed the spread operator wherever it appeared in the code

### PR DESCRIPTION
**Pull request by tigerhippo**
## Changes

I removed the spread operator wherever it appeared in the code. 

## Notes

I am not sure what to input for the fields projectLead and projectManager on lines 155 and 156 respectively. I did workPackage.wbsElement.projectLead and workPackage.wbsElement.projectManager (which should work because the WbsElement interface had all of these fields). However, the error message stated that the wbsElement did not have any of these fields.

I still need to test my code, but it would be great if someone could take a look at all the places where I removed the spread operator to see if I did it correctly!

### CODE REVIEW

File: src/backend/functions/projects.ts [hunk 1]
```diff
@@ -138,21 +145,33 @@ const projectTransformer = (
...
-        ...workPackage.wbsElement,
+        //...workPackage,
+        //...workPackage.wbsElement,
         id: workPackage.workPackageId,
         wbsNum: wbsNumOf(workPackage.wbsElement),
+        dateCreated: workPackage.wbsElement.dateCreated,
+        name: workPackage.wbsElement.name,
         status: convertStatus(workPackage.wbsElement.status),
+        projectLead: workPackage.wbsElement.projectLead,
+        projectManager: null,
```
Comment (jamescd18) [inline src/backend/functions/projects.ts:156]:
```suggestion
        projectManager: workPackage.wbsElement.projectManager,
```

File: src/backend/functions/projects.ts [hunk 2]
```diff
@@ -138,21 +145,33 @@ const projectTransformer = (
       );
 
       return {
-        ...workPackage,
-        ...workPackage.wbsElement,
+        //...workPackage,
+        //...workPackage.wbsElement,
```
Comment (jamescd18) [inline src/backend/functions/projects.ts:149]:
```suggestion
```

**Review by jamescd18 — CHANGES_REQUESTED** [review]
Looking good! Give those a try and attach screenshots of errors if you run into any. You can also still commit + push your code and see if the automated test runner will fail or not. If it doesn't, it's probably still good code.

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
great work! the only thing I want to add from James' review is that you committed the package-lock.json file which should not be committed except in very specific circumstances.

To uncommit this, you can essentially commit the current "main" version of the package-lock to your branch. This way, the package-lock will be the same as the one on main and then it wont be a change if that makes sense

**Comment by anthonybernardi** [comment]
i swear the package-lock is the exact same as the one on main. Go check line ~1496 for proof that the diff is wrong

**Comment by anthonybernardi** [comment]
image

this makes no sense and it's the same as on main so im merging this

**Review by anthonybernardi — APPROVED** [review]

**Pull request merged**